File: app/src/main/java/com/esorokin/lantector/ui/plugins/base/BaseDependencyPlugin.java
package com.esorokin.lantector.ui.plugins.base;

import android.content.Intent;
import android.support.annotation.NonNull;
import android.view.MenuItem;
import android.view.View;

public class BaseDependencyPlugin<T> extends DependencyPlugin<T> {
	public BaseDependencyPlugin(T dependency) {
		super(dependency);
	}

	@Override
	public void onCreate() {
		//override
	}

	@Override
	public void onViewCreated(final View view) {
		//override
	}

	@Override
	public void onStart() {
		//override
	}

	@Override
	public void onResume() {
		//override
	}

	@Override
	public void onPause() {
		//override
	}

	@Override
	public void onStop() {
		//override
	}

	@Override
	public void onDestroy() {
		//override
	}

	@Override
	public void onOptionsItemSelected(MenuItem item) {
		//override
	}

	@Override
	public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
		//override
	}

	@Override
	public void onActivityResult(int requestCode, int resultCode, Intent data) {
		//override
	}

	@Override
	public void onBackPressed() {
		//override
	}
}

File: app/src/main/java/com/esorokin/lantector/ui/plugin/base/BaseDependencyPlugin.kt
package com.esorokin.lantector.ui.plugin.base

import android.content.Intent
import android.view.MenuItem
import android.view.View

open class BaseDependencyPlugin<out T>(dependency: T) : DependencyPlugin<T>(dependency) {
    override fun onCreate() {
        //override
    }

    override fun onViewCreated(view: View) {
        //override
    }

    override fun onStart() {
        //override
    }

    override fun onResume() {
        //override
    }

    override fun onPause() {
        //override
    }

    override fun onStop() {
        //override
    }

    override fun onDestroy() {
        //override
    }

    override fun onOptionsItemSelected(item: MenuItem) {
        //override
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        //override
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        //override
    }

    override fun onBackPressed() {
        //override
    }
}